internal/cache/metadata/stat_cache.go

```diff
@@ -203,3 +219,46 @@ func (sc *statCacheBucketView) LookUp(
 
 	return
 }
+
+func (sc *statCacheBucketView) LookUpFolder(
+	folderName string,
+	now time.Time) (bool, *controlpb.Folder) {
+	// Look up in the LRU cache.
+	value := sc.sharedCache.LookUp(sc.key(folderName))
```

Comment:
Should we create common lookUp function which will return entry and then on respective functions we will assign folder or minObject?

Same for insert.

Reply:
e.g.
```
lookUp(return entry)
lookUpObject(set object from entry)
lookUpFolder(set folder from entry)
```
If it takes more effort then we can skip may be.

Reply:
Done for lookup, Insert is not having much repetitive code for now and so not refactoring insert for now.